plugins/woocommerce-admin/client/marketing/overview-multichannel/Campaigns/CreateNewCampaignModal/CreateNewCampaignModal.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment, useState } from '@wordpress/element';
+import {
+	Button,
+	CardDivider,
+	Modal,
+	Icon,
+	Flex,
+	FlexBlock,
+	FlexItem,
+} from '@wordpress/components';
+import { chevronUp, chevronDown, external } from '@wordpress/icons';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import {
+	useRecommendedChannels,
+	useNewCampaignTypes,
+	useRegisteredChannels,
+} from '~/marketing/hooks';
+import { SmartPluginCardBody } from '~/marketing/components';
+import './CreateNewCampaignModal.scss';
+
+const isExternalURL = ( url: string ) =>
+	new URL( url ).origin !== location.origin;
+
+/**
+ * Props for CreateNewCampaignModal, which is based on Modal.Props.
+ *
+ * Modal's title and children props are omitted because they are specified within the component
+ * and not needed to be specified by the consumer.
+ */
+type CreateCampaignModalProps = Omit< Modal.Props, 'title' | 'children' >;
+
+export const CreateNewCampaignModal = ( props: CreateCampaignModalProps ) => {
+	const { className, ...restProps } = props;
+	const [ collapsed, setCollapsed ] = useState( true );
+	const { data: newCampaignTypes } = useNewCampaignTypes();
+	const { data: recommendedChannels } = useRecommendedChannels();
+	const { refetch } = useRegisteredChannels();
+
+	return (
+		<Modal
+			className={ classnames(
+				className,
+				'woocommerce-marketing-create-campaign-modal'
+			) }
+			title={ __( 'Create a new campaign', 'woocommerce' ) }
+			{ ...restProps }
+		>
+			<div className="woocommerce-marketing-new-campaigns">
+				<div className="woocommerce-marketing-new-campaigns__question-label">
+					{ __(
+						'Where would you like to promote your products?',
+						'woocommerce'
+					) }
+				</div>
+				<div>
+					{ newCampaignTypes?.map( ( el ) => {
```

Comment:
Good catch. Not just in error situation, it could also happen when a registered channel does not provide any campaign types (also related to https://github.com/woocommerce/woocommerce/pull/37044#discussion_r1131035854).

I made a change in [`e4da8cb` (#37044)](https://github.com/woocommerce/woocommerce/pull/37044/commits/e4da8cba0976ea83eafd86c7033b5bf2c2d406cd), to display a "No campaign types found" message when there are no campaign types.

![image](https://user-images.githubusercontent.com/417342/224118533-898e7e08-8dd2-4b10-92f2-6a79a6e38607.png)